### ISSUE TITLE
Emit event when finished rendering

### ISSUE THREAD

**Issue by gone** [allain/x-widget#9]
I need some way to call htmx.process on the contents of the rendered component. The htmx docs have a great example for plain alpine.js at https://htmx.org/docs/#3rd-party  but it works by watching a variable.  It would be nice to emit an event on render with the id of the rendered object so I could globally catch the event and process the new nodes

**Comment by allain** [comment]
Makes sense. I can add that event for after when the widget's content is inserted into the page.

Children of it, which might be widgets themselves would not be rendered yet. Is that alright in your case?

**Comment by gone** [comment]
I think it would be fine as long as the children components also fire an event - I'd do something globally like:

```
document.addEventListener('x-widget-rendered', (event)=> { 
           htmx.process(event.target)
})
```

so I think that would work by firing once for each component recursively and  htmx would be made aware on each nested layer. 

I'm not familar enough with htmx's code to know for sure w/o trying though.

**Comment by allain** [comment]
Done in https://github.com/allain/x-widget/commit/878d3486c62ab3803ef7681e567520aff5e3835a

published in v0.2.8
